node.go

```diff
@@ -0,0 +1,75 @@
+package symbiosis
+
+import (
+	"fmt"
+)
+
+type Node struct {
+	ID       string `json:"id"`
+	Name     string `json:"name"`
+	NodeType struct {
+		ID        string `json:"id"`
+		Name      string `json:"name"`
+		MemoryMi  int    `json:"memoryMi"`
+		StorageGi int    `json:"storageGi"`
+		Vcpu      int    `json:"vcpu"`
+		Product   struct {
+			ProductCosts []struct {
+				Currency string  `json:"currency"`
+				UnitCost float32 `json:"unitCost"`
+			} `json:"productCosts"`
+		} `json:"product"`
+	} `json:"nodeType"`
+	Region struct {
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"region"`
+	PrivateIPv4Address string `json:"privateIPv4Address"`
+	State              string `json:"state"`
+	client             *Client
+}
+
+func (c *Client) DescribeNode(name string) (*Node, error) {
+	var result *Node
+
+	resp, err := c.symbiosisAPI.R().
+		SetResult(&result).
+		ForceContentType("application/json").
+		Get(fmt.Sprintf("rest/v1/node/%v", name))
+
+	if err != nil {
+		return nil, err
+	}
+
+	result.client = c
+	validated, err := c.ValidateResponse(resp, result)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if validated == nil {
+		return nil, nil
+	}
+
+	return validated.(*Node), nil
+}
+
+func (n *Node) Recycle() error {
```

Comment:
I think it's pretty clean to be able to do `node.Recycle()` instead of `client.Recycle(node.ID)`. But it also means that a node, or team member, has to be fetched in order to recycle, or delete it.

For example, the terraform provider only stores the ID of nodes, clusters, etc. I'm thinking if it's easier to use the client if all operations accept the ID instead of first fetching the full struct? It would also avoid having to inject the client into each object. I know this gets further complicated by the fact some routes use names though...

Reply:
@Styren 

Wdyt of this:

```
member, err := client.Team.Member.GetByEmail("stephen@limit404.com")
team, err := client.Team.List()
```

OR

```
member, err := client.Team.TeamMember.GetByEmail("stephen@limit404.com")
team, err := client.Team.List()
```

Reply:
I opted to go for 

```
client.Team.GetMemberByEmail
client.Team.GetInvitationByEmail
client.Team.InviteMembers
client.Team.DeleteMember
```

Etc

Reply:
> I opted to go for
> 
> ```
> client.Team.GetMemberByEmail
> client.Team.GetInvitationByEmail
> client.Team.InviteMembers
> client.Team.DeleteMember
> ```
> 
> Etc

Looks great! 🙌